src/main/scala/firrtl/AddDescriptionNodes.scala

```diff
@@ -6,7 +6,11 @@ import firrtl.ir._
 import firrtl.annotations._
 import firrtl.Mappers._
 
-case class DescriptionAnnotation(named: Named, description: String) extends Annotation {
+sealed trait DescriptionType
+case object DocStringDescription extends DescriptionType
+case object AttributeDescription extends DescriptionType
+
+case class DescriptionAnnotation(named: Named, description: String, tpe: DescriptionType) extends Annotation {
```

Comment:
This seems to be a visible API feature. Shouldn't this at least have a default value for `tpe` so it won't break potential uses in the wild?

Reply:
Fair point. I'm now wondering if it would be better to have different annotations for each `DescriptionType` rather than a `tpe` field. But in that case, I think I'd want to change the name of `DescriptionAnnotation`